soc: riscv: andes_v5: fix missing header in PMA driver and the MPU padding issue in linker.ld

jimmyzhe opened the pull request:
This PR fix two issues in the andes_v5 SoC

1. PMA driver lacks the `#include "soc_v5.h"` for CSR definition.
2. In the linker.ld, the location counter is not correct for padding the MPU region because it had been assigned to `RAM_BASE`. Fixed this issue by utilizing LMA.

jimmyzhe committed:
soc: andestech: pma.c: include soc_v5.h

Add include header for CSR definition.

Signed-off-by: Jimmy Zheng <jimmyzhe@andestech.com>

jimmyzhe committed:
soc: andestech: linker.ld: fixed ROM padding issue

Updated ROM region padding to utilize LMA instead of location counter,
because the location counter has been assigned to 'RAM_BASE'.

Signed-off-by: Jimmy Zheng <jimmyzhe@andestech.com>

jimmyzhe commented:
Rebased.

edersondisouza commented on soc/andestech/ae350/linker.ld:
```diff
@@ -423,8 +423,8 @@ __rom_region_end = LOADADDR(.last_section) + SIZEOF(.last_section);
 	MPU_ALIGN(__rom_region_size);
-	__rom_region_end = .;
 } GROUP_LINK_IN(ROMABLE_REGION)
+__rom_region_end = LOADADDR(rom_mpu_padding) + SIZEOF(rom_mpu_padding);
```
It seems comment on line 403 is being contradicted by this change - maybe update the comment as well?

jimmyzhe replied:
Thanks for your suggestion. 

I think the comment on line 403 still applies to this change.
`__rom_region_end`  is still at the end of the linker script, but the last section is `rom_mpu_padding` instead of  `.last_section`.

I moved this comment to the assignment of `__rom_region_end` to avoid confusion.
```
/* Because ROMABLE_REGION != RAMABLE_REGION in XIP-system, it is valid
 * to set __rom_region_end symbol at the end of linker script and
 * doesn't mistakenly contain the RAMABLE_REGION in it.
 */
```

jimmyzhe committed:
soc: andestech: linker.ld: clarify usage of __rom_region_end/size

Clarify usage of __rom_region_end/size in XIP system.
When PMP is enabled, __rom_region_end should be padded to meet the
requirement of PMP entry, and the actual ROM region usage ends at
.last_section instead of __rom_region_end.

Signed-off-by: Jimmy Zheng <jimmyzhe@andestech.com>

jimmyzhe commented:
I also updated the comment about `__rom_region_size` because it no longer represents the size of the image after `rom_mpu_padding`.

I think the size of the image can be represented by another symbol in the future, like `_flash_used` in arm. 
https://github.com/zephyrproject-rtos/zephyr/blob/db237875376e3ef548f3691994a7fbb35caef254/include/zephyr/arch/arm/cortex_m/scripts/linker.ld#L487
